### ISSUE TITLE
Infinity redirect after login

### ISSUE THREAD

**Issue by mrmartineau** [supabase-community/supabase-auth-helpers#94]
Since upgrading to v1.4.0, when successfully signing in, my Network tab goes mental alternating between my sign in route (`/signin`) and the route that we redirect to straight after sign in (`/feed`). Presumably the check if a user is authenticated (after signin) fails and redirects back to signin, where another check occurs which says they're authenticated, and redirects again ad infinitum.

<img width="1305" alt="image" src="https://user-images.githubusercontent.com/64883/167777067-4b76cf5a-c942-4af7-8bab-0c53559844b1.png">

I will try to provide more info but I'm using this package on a side project and get very little time to develop it

FYI I am using the new `withPageAuth` from the 1.4.0 release.

Is anyone else experiencing this issue?

**Comment by thorwebdev** [comment]
Can you please provide reproduction steps and your code?

**Comment by thorwebdev** [comment]
You will need to make sure to only redirect once the user in the `useUser` hook becomes available. See an example here: https://github.com/vercel/nextjs-subscription-payments/blob/main/pages/signin.tsx#L58-L62

**Comment by mrmartineau** [comment]
> Can you please provide reproduction steps and your code?
It is very unlikely that I'll have time to do that I'm afraid, but... for what it's worth, I just reverted to v1.3.1 and the issue is resolved (I had to revert to use `withAuthRequired` of course too).

I then thought that if I used `withAuthRequired` with v1.4.0 that it might work, alas it did not.

What changed in those functions between the two versions? or could it be related to a dependency or one of the auth API routes?

**Comment by thorwebdev** [comment]
`withPageAuth` works as expected for me on https://github.com/vercel/nextjs-subscription-payments, so I'm afraid without code examples / reproduction steps I won't be able to dig into this.

**Comment by mrmartineau** [comment]
So part of the reason for the redirect is because I have some logic that redirects the user when they signed in, from pages that should only be accessed when signed out, back to the app. e.g. the `/signin` or `/register` pages should not be accessed when the user is signed in, so if a user visits one of those pages, they will be redirected.

The logic in `getServerSideProps` looks like this:

```ts
import type { GetServerSideProps, NextPage } from 'next'
import { AuthForm } from '../components'

const Signin: NextPage = () => {
  return <AuthForm type="signin" />
}

export const getServerSideProps: GetServerSideProps = async (context) => {
  const access_token = context.req.cookies[`sb-access-token`]
  if (access_token) {
    return {
      redirect: {
        destination: '/app',
        permanent: false,
      },
    }
  }
  return {
    props: {},
  }
}

export default Signin
```

This causes the redirect loop, but only because after a successful login the user is being redirected back to `/signin`. If I remove the `getServerSideProps` from the `Signin` pages component the loop stops, but as previously stated the user is redirected back to `/signin` because somewhere in the `withPageAuth` logic something an exception is thrown which triggers the redirect.

FYI I am editing the `withPageAuth` locally to see at what point the exception is thrown, so I will come back with any findings.

I have just bumped to 1.4.1 to see if that fixed anything - it did not.

**Comment by mrmartineau** [comment]
Ok, so I have just figured out why this bug started happening:`atob`, like in #93 but I did not see any errors when running `next dev` locally. 

I was running node 14.x locally because a client project required it and since updating to node 16.x and supabase-auth-helpers everything is fine. This was driving me nuts because there was no error reporting and I hadn't deployed to Vercel since these errors started so I didn't associate my bug with #93. 

FYI @thorwebdev
